### ISSUE TITLE
fix(training): LRScheduler wired after CheckpointLoader resumes from already-decayed base_lrs

### ISSUE THREAD

**Issue by treeleaves30760** [treeleaves30760/CodefyUI#149]
> 中文摘要:若圖的拓撲順序是 CheckpointLoader 先還原 optimizer(LR 已衰減)、LRScheduler 才據此建立,scheduler 的 base_lrs 就是衰減後的值,#118 的快轉重放會從錯的基準出發。checkpoint 內含 scheduler 狀態時可避開;無狀態舊檔時需另行處理。

## Context
Flagged by the #118 implementer (report concern 1): the start_epoch fast-forward assumes the scheduler was built at the run's original LR. If LRSchedulerNode constructs from an optimizer whose LR was already restored to a decayed value, base_lrs are decayed and unrecoverable. Mitigated when the checkpoint carries scheduler_state_dict (that route takes precedence); the gap is legacy checkpoints without it.

## Sketch
Options: store original base_lrs in the checkpoint payload; or have LRSchedulerNode warn when built from an optimizer whose param-group lr differs from defaults; or document the wiring rule.

Refs: #118, PR #147.

**Comment by treeleaves30760** [comment]
## Re-diagnosis: the stated cause does not reproduce

> 中文摘要：這個 issue 原本寫的成因是錯的。實測兩種接法的 `base_lrs` 都是 0.1，跟理論 cosine 相符到 1.4e-17；`initial_lr` 會跟著 `optimizer.state_dict()` 一起被存下來再還原，而 `LRScheduler.__init__` 是用 `setdefault` 讀它，不會覆蓋。原本寫的「缺 scheduler_state_dict 的舊檔就是破口」也不成立（實測 OK）。真正會出事的條件是 checkpoint 裡**缺 `initial_lr`**，而本產品在有排程器的執行下不會寫出這種檔。不過在這個接法上確實有一個**真實可觸發**的缺陷，只是跟 `base_lrs` 無關 — 見下方第 3 節。

This came up while validating core#138 (PR #206), where both docs pages had repeated the mechanism from this issue body. I ran it through the real nodes rather than reasoning about it, and the mechanism does not hold. Posting the measurements so whoever picks this up does not start from the wrong model.

### 1. `base_lrs` is not decayed, in either wiring order

Both orders through the real `OptimizerNode` / `LRSchedulerNode` / `CheckpointLoaderNode` / `_prepare_scheduler`, SGD lr=0.1, `CosineAnnealingLR(T_max=20)`, checkpoint taken at epoch 8, resumed to epoch 20 and compared against the closed-form cosine:

```
Order A  Optimizer -> LRScheduler -> CheckpointLoader.lr_scheduler
         base_lrs=[0.1]  first resumed lr=0.0654508497  max|lr - theory| = 1.388e-17

Order B  Optimizer -> CheckpointLoader -> LRScheduler   (loader.lr_scheduler unwired)
         base_lrs=[0.1]  first resumed lr=0.0654508497  max|lr - theory| = 1.388e-17
```

The reason is two lines of PyTorch (verified in 2.11.0+cu128):

- `Optimizer.state_dict()` packs **every** param-group key except `params`:
  `packed = {k: v for k, v in group.items() if k != "params"}`. `initial_lr`, which the
  original scheduler stamped onto the group, is therefore saved and restored.
- `LRScheduler.__init__` reads it with `group.setdefault("initial_lr", initial_lr)` —
  `setdefault` does **not** overwrite an existing value, and `base_lrs` is then read from
  `initial_lr`, not from `lr`.

So a scheduler constructed on an already-restored optimizer still starts from 0.1. Direct check:

```
after optimizer.load_state_dict(checkpoint):  lr=0.0654508497  initial_lr=0.1
scheduler built on that restored optimizer:   base_lrs=[0.1]
```

### 2. The "legacy checkpoints without scheduler_state_dict" gap is also not the trigger

Full vintage matrix, same harness:

| checkpoint vintage | Order A | Order B |
|---|---|---|
| modern (`scheduler_state_dict` + `initial_lr`) | OK, 1.4e-17 | OK, 1.4e-17 |
| **legacy pre-#118 (no `scheduler_state_dict`, `initial_lr` present)** | **OK, 1.4e-17** | **OK, 1.4e-17** |
| no `initial_lr` only | OK, 1.4e-17 | DEVIATES 2.26e-02, `base_lrs=[0.065451]` |
| neither | OK, 1.4e-17 | DEVIATES 2.26e-02 |

Row 2 is the case this issue names as the gap, and it is fine — `#118`'s fast-forward replays from `base_lrs`, and `base_lrs` is correct because `initial_lr` survived inside the optimizer state. The discriminator is **`initial_lr`**, not `scheduler_state_dict` and not the wiring order.

And a checkpoint missing `initial_lr` is not something this product writes from a scheduled run. Constructing the scheduler stamps `initial_lr` onto the live optimizer, so it is in `optimizer.state_dict()` even when `CheckpointSaver.lr_scheduler` is left unwired:

```
LRScheduler wired to TrainingLoop but NOT into CheckpointSaver:
    scheduler_state_dict present: False
    initial_lr present:           0.1      <- still there
graph with no LRScheduler at all:
    initial_lr present:           None     <- but there is no schedule to resume
```

**As stated, this issue describes an unreachable failure.**

### 3. There IS a real defect in this wiring, and it is a different one

Leaving `CheckpointLoader.lr_scheduler` unwired (it is an optional input, `checkpoint_node.py:102-107`) means the loader has nothing to restore `scheduler_state_dict` **into**, so it discards it with a log line and the schedule is reconstructed by `_prepare_scheduler` replaying `start_epoch` steps from `base_lrs`.

For `CosineAnnealingLR` the replay is exact, which is why nothing has been observed to break. For a metric-driven schedule it cannot be — and `_fast_forward_scheduler` already says so in a warning:

> `ReduceLROnPlateau is driven by the validation metric and cannot be replayed from start_epoch=%d. Wire the scheduler through CheckpointSaver/CheckpointLoader to restore its state exactly.`

Measured, `ReduceLROnPlateau` resumed from an 8-epoch checkpoint whose last 5 epochs were a plateau:

```
writer at epoch 8:  best=0.8  num_bad_epochs=5

Order A (scheduler wired into the loader):
    best=0.8  num_bad_epochs=5  last_epoch=8      <- restored exactly
    after one more plateau epoch: num_bad_epochs=6

Order B (loader.lr_scheduler unwired):
    best=inf  num_bad_epochs=0  last_epoch=0      <- plateau history silently lost
    after one more plateau epoch: num_bad_epochs=0
```

The decay that was one epoch away is postponed indefinitely, with no error and only an INFO log.

### Suggested restatement

The title and body should be re-scoped to something like: *"CheckpointLoader silently discards `scheduler_state_dict` when `lr_scheduler` is left unwired"*. Concretely, the fix candidates change too:

- The "store original `base_lrs` in the checkpoint payload" option in the sketch is unnecessary — `initial_lr` already round-trips.
- The "warn when built from an optimizer whose param-group lr differs from defaults" option would fire on a perfectly correct resume.
- What would actually help: have `CheckpointLoader` **warn** (not INFO) when a checkpoint carries `scheduler_state_dict` but no scheduler is wired in — it currently logs at INFO (`checkpoint_node.py:222-226`) — and/or have `_prepare_scheduler` refuse to silently replay a `ReduceLROnPlateau`.

Docs in PR #206 have been corrected to state this mechanism rather than the original one, in both locales.

Reproduction harness: both orders driven through the shipped nodes on torch 2.11.0+cu128; happy to attach the script if useful.
